packages/nextjs/app/_components/stats/LargeGrantsStats.tsx

```diff
@@ -0,0 +1,30 @@
+import { StatsItem } from "./StatsItem";
+import { getPublicLargeGrants } from "~~/services/database/repositories/large-grants";
+
+export const LargeGrantsStats = async () => {
+  const allLargeGrants = await getPublicLargeGrants();
+  const largeGrants = allLargeGrants.filter(grant => {
+    const latestStage = grant.stages[0];
+    return (
+      (latestStage.status !== "proposed" && latestStage.status !== "rejected" && latestStage.stageNumber === 1) ||
+      latestStage.stageNumber !== 1
+    );
+  });
+
+  const usdcGranted = largeGrants
```

Comment:
Maybe we can get this information with a query to the DB, avoiding looping all the grants.

Reply:
Optimized query for large grants [6eff33d](https://github.com/BuidlGuidl/ens-pg/pull/32/commits/6eff33d0ea7bdfbd600186ead68641533d46fb93)

Tried to do the same for regular grants but ai couldn't find better solution because of this
```
const latestStage = grant.stages[0];
    return (
      (latestStage.status !== "proposed" && latestStage.status !== "rejected" && latestStage.stageNumber === 1) ||
      latestStage.stageNumber !== 1
    );
``` 
and it suggest to do almost the same